--- a/configure.ac
+++ b/configure.ac
@@ -83,7 +83,7 @@
 
 dnl Checks for libraries.
 dnl AC_CHECK_LIB([pthread], [pthread_create])  # should use ast_pthread_join everywhere?
-AC_SEARCH_LIBS([iconv], [c iconv],,AC_MSG_ERROR([iconv library missing]))
+AC_SEARCH_LIBS([libiconv], [iconv],,AC_MSG_ERROR([iconv library missing]))
 AC_CHECK_LIB([sqlite3], [sqlite3_open],,AC_MSG_ERROR([sqlite3 library missing]))
 
 AC_DEFINE([ICONV_CONST],[], [Define to const if you has iconv() const declaration of input buffer])
  AC_MSG_CHECKING([for iconv use const inbuf])
@@ -224,8 +225,6 @@
 if test "x$enable_debug" = "xyes" ; then
   CFLAGS="$CFLAGS -O0 -g"
   AC_DEFINE([__DEBUG__], [1], [Build with debugging])
-else
-  CFLAGS="$CFLAGS -O6"
 fi
 
 dnl Asterisk header files use lots of old style declarations, ignore those.
